public class Leetcode_704 {

    public int search(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;

        while (left <= right) {

            
            int mid = left + (right - left) / 2;

            if (nums[mid] == target) {
                return mid;  
            } 
            else if (nums[mid] < target) {
                left = mid + 1; 
            } 
            else {
                right = mid - 1; 
            }
        }

        return -1;  
    }

 
    public static void main(String[] args) {
        Leetcode_704 obj = new Leetcode_704();

        int[] nums = {-1, 0, 3, 5, 9, 12};
        int target = 3;

        int result = obj.search(nums, target);

        System.out.println("Index of target: " + result);
    }
}